nltk/stem/arlstem.py

```diff
@@ -213,54 +213,57 @@ def fem2masc(self, token):
         """
             transform the word from the feminine form to the masculine form.
         """
-        if(token.endswith('\u0629') and len(token) > 3):
+        if token.endswith('\u0629') and len(token) > 3:
             token = token[:-1]
             return token
 
     def plur2sing(self, token):
         """
             transform the word from the plural form to the singular form.
         """
-        if(len(token) > 4):
+        if len(token) > 4:
             for ps2 in self.pl_si2:
-                if(token.endswith(ps2)):
+                if token.endswith(ps2):
                     token = token[:-2]
                     return token
-        if(len(token) > 5):
+        if len(token) > 5:
             for ps3 in self.pl_si3:
-                if(token.endswith(ps3)):
+                if token.endswith(ps3):
                     token = token[:-3]
                     return token
-        if(len(token) > 3 and token.endswith('\u0627\u062A')):
+        if len(token) > 3 and token.endswith('\u0627\u062A'):
             token = token[:-2]
             return token
         if(len(token) > 3 and token.startswith('\u0627')
-           and token[2] == '\u0627'):
-            token = token[0:2] + token[3:]
+           and token[2] == '\u0627'
+           ):
+            token = token[:2] + token[3:]
             return token
         if(len(token) > 4 and token.startswith('\u0627')
-           and token[-2] == '\u0627'):
-            token = token[1:]
-            token = token[:-2] + token[-1]
+           and token[-2] == '\u0627'
+           ):
+            token = token[1:-2] + token[-1]
             return token
 
     def verb(self, token):
         """
             stem the verb prefixes and suffixes or both
         """
-        vb = self.verb_t1(token)
-        if vb is None:
-            vb = self.verb_t2(token)
+        try:
+            vb = self.verb_t1(token)
             if vb is None:
-                vb = self.verb_t3(token)
+                vb = self.verb_t2(token)
                 if vb is None:
-                    vb = self.verb_t4(token)
+                    vb = self.verb_t3(token)
                     if vb is None:
-                        vb = self.verb_t5(token)
-                        if vb is not None:
-                            token = vb
+                        vb = self.verb_t4(token)
+                        if vb is None:
+                            vb = self.verb_t5(token)
+                            if vb is not None:
```

Comment:
Please try the following and let's see how it looks:

* get your `verb_tn` functions to raise an exception if passed the empty string
* run through them all in the calling program without any nested conditionals

Reply:
Thanks for the valuable suggestion ! Nested conditions are removed as well as I have added an exception (to test empty string) in stem function